Board.py

```diff
@@ -1,11 +1,163 @@
-from checker import Checker
+from enum import Enum
+
+ROWS = 8
+COLS = 8
+
+
+class Player(Enum):
+    player1 = 0
+    player2 = 1
+
+
+class Piece(Enum):
+    basic = 0
+    king = 0
+
+
+def in_bounds(x, y):
+    return 0 <= x < COLS and 0 <= y < ROWS
+
+
+def is_jump(start_x, end_x):
+    return abs(end_x - start_x) > 1
 
 
 class Board:
+
     def __init__(self):
-        rows, cols = 8, 8
-        self.board = [['0' for _ in range(cols)] for _ in range(rows)]
+        self.player_one_pieces = []
+        self.player_two_pieces = []
+        for i in range(3):
+            for z in range(4):
+                checker_x = i
+                checker_y = z * 2 + (i + 1) % 2
+                temp_piece = self.generate_checker(checker_x, checker_y, Player.player1)
+                self.player_one_pieces.append(temp_piece)
+
+                checker_x = i + 5
+                checker_y = z * 2 + i % 2
+                temp_piece = self.generate_checker(checker_x, checker_y, Player.player2)
+                self.player_two_pieces.append(temp_piece)
+
+    def move(self, move):  # move tuple ==> (start_x, start_y, end_x, end_y, Player)
+        start_x = move[0]
+        start_y = move[1]
+        end_x = move[2]
+        end_y = move[3]
+        player = move[4]
+
+        if is_jump(start_x, end_x):
+            self.remove_checker(int((start_x + end_x) / 2), int((start_y + end_y) / 2))  # remove piece
+
+        self.remove_checker(start_x, start_y)
+        new_checker = self.generate_checker(end_x, end_y, player)
+        self.add_new_checker(new_checker)
+
+    def generate_checker(self, x, y, player): # checker structure ==> (x, y, Player, Piece)
+        if y == 0 or y == ROWS:
+            return x, y, player, Piece.king
+        else:
+            return x, y, player, Piece.basic
+
+    def add_new_checker(self, checker):
+        if checker[3] == Player.player1:
```

Comment:
for readability, i would personally add a couple helper functions like `get_player_from_piece(checker)` or `get_type_from_piece(checker)` which are one line helper functions that make code more readable